### PR TITLE
feat: Add support for the ion shell

### LINKED ISSUE #678: Add support for the ion shell
#### Is your feature request related to a problem? Please describe.

Currently starship doesn't support the [ion shell](https://gitlab.redox-os.org/redox-os/ion/). It would be awesome if we could add support for this.

#### Describe the solution you'd like

It'd be awesome if we could use a similar setup to what other shells have. So for example being able to just add `eval "$(starship init ion)"` or something similar.

I think it'd be interested in trying to implement this if someone could point me in the right direction. I have only some basic experience with rust, but I think I can figure this out.

#### Describe alternatives you've considered

Currently it is possible to get it to kind of work with the ion shell. Using the following `PROMPT` function in the `initrc` configuration file:

```ion
fn PROMPT
	starship prompt --status=$? --jobs=$(jobs ^| wc -l)
end

export STARSHIP_SHELL="ion"
```

The `export STARSHIP_SHELL="ion"` is needed, because of issue #211.

### PR DESCRIPTION
**Pull request by BuggStream**
#### Description

This commit adds support for the jobs and status options of the starship prompt.
Support for the cmd-duration is also added if you are using the latest
ion shell. The cmd-duration is implemented using the ION_CMD_DURATION variable.
This variable keeps track of the time spent executing the last command. (This is
similar to the CMD_DURATION of the fish shell).

See this MR for more information: https://gitlab.redox-os.org/redox-os/ion/merge_requests/1176

#### Motivation and Context

Closes #678 

This change is useful for anyone who likes to use ion and starship together.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):
image

#### How Has This Been Tested?

- [ ] ~~I have tested using **MacOS**~~
- [x] I have tested using **Linux**
- [ ] ~~I have tested using **Windows**~~

The ion shell is only supported on linux and redox-os.

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

I have updated the documentation and README with references to ion. But I haven't updated the translations, since I don't speak those other languages.

I also haven't added any tests, since I couldn't find any tests that test all the different shells. If there are such tests, let me know and I will try to add them.

### CODE REVIEW

File: src/init/mod.rs [hunk 1]
```diff
@@ -109,6 +109,10 @@ fi"#,
             );
             Some(script)
         }
+        Some("ion") => {
+            let script = format!("eval $({} init ion --print-full-init)", starship);
+            Some(script)
+        }
```
Comment (Scotsguy) [inline src/init/mod.rs]: Can't comment on the relevant part, but maybe change L.129: "For the time being, we support bash, zsh and fish."?

Comment (BuggStream) [reply src/init/mod.rs]: I added a reference to `ion` on the line you mentioned and also to the install.sh script. I think this should be all of it (ignoring the documentation in other languages).

**Comment by charleschege** [comment]
Can we have this pull request merged?

**Comment by BuggStream** [comment]
> Can we have this pull request merged?

As far as I am aware everything has been completed.

**Comment by matchai** [comment]
@all-contributors Please add @BuggStream for code and docs. 🎉

**Pull request merged**